internal/cli-flags/Parser.ts

```
@@ -1141,41 +1159,53 @@ export default class Parser<T> {
 
 		// If we don't have a suggestion then try to find another closest one
 		if (opts.suggestedName === undefined) {
-			opts.suggestedName = findClosestStringMatch(
+			const suggestions = findClosestMatchingStrings(
 				commandName,
 				Array.from(this.commands.keys()),
+				0.3,
 			);
-		}
 
-		// Set suggestedCommand
-		if (opts.suggestedName !== undefined) {
-			opts.suggestedCommand = serializeCLIFlags(
-				{
-					...this.getSerializeOptions(),
-					commandName: opts.suggestedName,
-					args: displayArgs,
-					defaultFlags,
-					flags: rawFlags,
-				},
-				"none",
-			).sourceText;
-		}
+			if (suggestions.length > 1) {
+				let options: SelectOptions = {};
 
-		const diag: Diagnostic = {
-			description: descriptions.FLAGS.UNKNOWN_COMMAND(opts),
-			location: serializeCLIFlags(
-				{
-					...this.getSerializeOptions(),
-					commandName,
-					args: displayArgs,
-					defaultFlags,
-					flags: rawFlags,
-				},
-				"command",
-			),
-		};
+				suggestions.forEach((suggestion) => {
+					options[suggestion] = {label: markup`${suggestion}`};
+				});
 
-		throw new DiagnosticsError("Unknown command", [diag]);
+				const selected = await this.reporter.select(
+					markup`Unknown command. Found matching commands: `,
+					{
+						radio: true,
+						options,
+					},
+				);
+
+				const args = Array.from(selected)[0].split(" ");
+				await this.reinit(args);
+			} else if (suggestions.length === 1) {
+				const yes = await this.reporter.radioConfirm(
+					markup`Did you mean ${suggestions[0]}`,
+				);
+				if (yes) {
+					await this.reinit(suggestions);
+				}
```

What would happen if the user didn't confirm?

---

Should probably fall back to `throw new DiagnosticsError("Unknown command", [diag]);`